### PR TITLE
CODEOWNERS/MAINTAINERS: Add soc/Kconfig ownership

### PR DESCRIPTION
**Pull request by nordicjm**
Adds soc/Kconfig ownership to @tejlmand and @nordicjm for CODEOWNERS file and build system in MAINTAINERS file.

Issue noticed due to a PR touching this file and having no-one assigned to it.

### CODE REVIEW

File: MAINTAINERS.yml [hunk 1]
```diff
@@ -321,6 +321,7 @@ Build system:
     - doc/develop/modules.rst
     - scripts/build/
     - tests/cmake/
+    - soc/Kconfig
```
[the remark this answers is not in the document]
Comment (nashif) [reply MAINTAINERS.yml:324]: an area for "Board / SoC Configuration" makes sense. I do not think this is part of the build system.

[the remark this answers is not in the document]
Comment (galak) [reply MAINTAINERS.yml:324]: agreed, this felt weird being put under Build System.

[the remark this answers is not in the document]
Comment (nordicjm) [reply MAINTAINERS.yml]: Updated and added @galak to list

**Review by stephanosio — DISMISSED** [review]

**Review by MaureenHelm — DISMISSED** [review]

File: MAINTAINERS.yml [hunk 2]
```diff
@@ -324,6 +324,19 @@ Build system:
...
+Board/SoC configuration:
+  status: maintained
+  maintainers:
+    - tejlmand
+  collaborators:
+    - galak
+    - nashif
+    - nordicjm
+  files:
+    - soc/Kconfig
```
Comment (nashif) [inline MAINTAINERS.yml:336]: what about other files? i.e. boards/Kconfig

**Review by nashif — CHANGES_REQUESTED** [review]

File: MAINTAINERS.yml (hunk 2 above)
Comment (nordicjm) [reply MAINTAINERS.yml:336]: Fixed

**Review by stephanosio — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**